docs/how-to/h-setup-k8s.md

```diff
@@ -0,0 +1,163 @@
+
+```bash 
+sudo snap alias microk8s.kubectl kubectl
+sudo usermod -a -G microk8s ${USER}
+mkdir -p ~/.kube
+sudo chown -f -R ${USER} ~/.kube
+```
+
+To make these changes effective, you can either open a new shell (or log in, log out) or use `newgrp microk8s` 
+
+Make sure that the MicroK8s cluster is now up and running:
+
+```bash
+microk8s status --wait-ready
+```
+
+Export the Kubernetes config file associated with admin rights and store it in the $KUBECONFIG file, e.g. `~/.kube/config`: 
+
+```bash 
+export KUBECONFIG=path/to/file # Usually ~/.kube/config
+microk8s config | tee ${KUBECONFIG}
+```
+
+Enable the K8s features required by the Spark Client Snap 
+
+```
+microk8s.enable dns rbac storage hostpath-storage
+```
+
+#### S3 Storage
+
+Enable the MinIO storage if you want to store Spark Jobs logs in a S3 compatible object storage to be then exposed via Charmed Spark History Server.
+
+```
+microk8s.enable minio
+```
+
+Refer [here](https://microk8s.io/docs/addon-minio) for more information how to customize your MinIO Microk8s deployment.
+
+Use the following commands to obtain the access key, the access secret and the MinIO endpoint:
+
+* *access key*: `microk8s.kubectl get secret -n minio-operator microk8s-user-1 -o jsonpath='{.data.CONSOLE_ACCESS_KEY}' | base64 -d`
+* *secret key*: `microk8s.kubectl get secret -n minio-operator microk8s-user-1 -o jsonpath='{.data.CONSOLE_SECRET_KEY}' | base64 -d`
+* *MinIO endpoint*: `microk8s.kubectl get services -n minio-operator | grep minio | awk '{ print $3 }'`
+
+#### External LoadBalancer
+
+Enable external loadbalancer if you want to expose the Spark History Server UI via a Traefik ingress
+
+```
+IPADDR=$(ip -4 -j route get 2.2.2.2 | jq -r '.[] | .prefsrc')
+microk8s enable metallb:$IPADDR-$IPADDR
+```
+
+The MicroK8s cluster is now ready to be used. 
+
+
+### AWS EKS
+
+In order to deploy an EKS cluster, make sure that you have working CLI tools properly installed on your edge machine:
+
+* [AWS CLI](https://aws.amazon.com/cli/) correctly setup to use a properly configured service account (refer to [here](https://docs.aws.amazon.com/cli/latest/userguide/cli-chap-configure.html) for configuration and [here](https://docs.aws.amazon.com/cli/latest/userguide/cli-chap-authentication.html) for authentication). Once the AWS CLI is configured, make sure that it works properly by testing the authentication call through the following command:
+```bash 
+aws sts get-identity-caller
+```
+
+* [eksctl](https://eksctl.io/) installed and configure (refer to the [README.md] file for more information on how to install it)
+
+Make also sure that your service account (configured in AWS) has the right permission to create and manage EKS clusters. In general, we recommend the use of profiles when having multiple accounts.
+
+#### Creating the cluster
+
+Create a YAML file with the following content 
```

Comment:
Maybe mention that there are several ways to create a cluster, for instance via the AWS console or with a tool like Terraform, as well as via the CLI, but for this tutorial we'll use the CLI.